packages/mobile/src/paymentRequest/PaymentRequestConfirmation.tsx

```diff
@@ -23,86 +22,69 @@ import { StackParamList } from 'src/navigator/types'
 import { writePaymentRequest } from 'src/paymentRequest/actions'
 import { PaymentRequestStatus } from 'src/paymentRequest/types'
 import { getDisplayName } from 'src/recipients/recipient'
-import { RootState } from 'src/redux/reducers'
-import { ConfirmationInput, getConfirmationInput } from 'src/send/utils'
+import useSelector from 'src/redux/useSelector'
+import { getConfirmationInput } from 'src/send/utils'
 import DisconnectBanner from 'src/shared/DisconnectBanner'
 import Logger from 'src/utils/Logger'
-import { currentAccountSelector } from 'src/web3/selectors'
+import { walletAddressSelector } from 'src/web3/selectors'
 
 // @ts-ignore
 const TAG = 'paymentRequest/confirmation'
 
-interface StateProps {
-  e164PhoneNumber: string | null
-  account: string | null
-  confirmationInput: ConfirmationInput
-  addressJustValidated?: boolean
-}
+type OwnProps = StackScreenProps<StackParamList, Screens.PaymentRequestConfirmation>
 
-interface DispatchProps {
-  showError: typeof showError
-  writePaymentRequest: typeof writePaymentRequest
-}
+type Props = WithTranslation & OwnProps
+
+export const paymentConfirmationScreenNavOptions = () => ({
+  ...emptyHeader,
+  headerLeft: () => <BackButton eventName={RequestEvents.request_confirm_back} />,
+})
 
-const mapDispatchToProps = { showError, writePaymentRequest }
+function PaymentRequestConfirmation({ route }: Props) {
+  const [comment, setComment] = useState('')
+  const [transactionData] = useState(route.params.transactionData)
+  const [addressJustValidated] = useState(route.params.addressJustValidated)
+
+  const e164NumberToAddress = useSelector((state) => state.identity.e164NumberToAddress)
+  const secureSendPhoneNumberMapping = useSelector(
+    (state) => state.identity.secureSendPhoneNumberMapping
+  )
+  const account = useSelector(walletAddressSelector)
```

Comment:
small nit: I would rename for clarity `account` to `walletAddress` since `account` can be confused with the MTW. I know some of these comments are of preexisting code but I figured we could add these small clarity improvements since we're at it :)